### PR TITLE
SPCTR-3142: Fix - Inline Notice default padding issue in editor 

### PR DESCRIPTION
**Pull request by rajmohanreddy774**
### Description

Inline notice content padding in editor is different from frontend.
https://brainstormforce.atlassian.net/browse/SPCTR-3142

### Screenshots

https://d.pr/v/Yp2tIX
### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [ ] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by AbhijitNage123 — APPROVED** [review]

**Review by bsf-zanev — COMMENTED** [review]
Just a few comments.

File: README.md [hunk 1]
```diff
@@ -215,6 +215,9 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 
 ## Changelog ##
 
+### x.x.x ###
+* Fix: Inline Notice - Resolved issues where content doesn't have default padding in editor as frontend and removed unwanted margin on top of content in frontend.
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply README.md]: Updated.

File: includes/blocks/inline-notice/frontend.css.php [hunk 2]
```diff
@@ -36,10 +36,10 @@
 $title_left_padding_tablet   = isset( $attr['titleLeftPaddingTablet'] ) ? $attr['titleLeftPaddingTablet'] : $attr['titleHrPadding'];
 $title_right_padding_tablet  = isset( $attr['titleRightPaddingTablet'] ) ? $attr['titleRightPaddingTablet'] : $attr['titleHrPadding'];
 
-$content_top_padding    = isset( $attr['contentTopPadding'] ) ? $attr['contentTopPadding'] : $attr['contentVrPadding'];
-$content_bottom_padding = isset( $attr['contentBottomPadding'] ) ? $attr['contentBottomPadding'] : $attr['contentVrPadding'];
-$content_left_padding   = isset( $attr['contentLeftPadding'] ) ? $attr['contentLeftPadding'] : $attr['contentHrPadding'];
-$content_right_padding  = isset( $attr['contentRightPadding'] ) ? $attr['contentRightPadding'] : $attr['contentHrPadding'];
+$content_top_padding    = ! empty( $attr['contentTopPadding'] ) ? $attr['contentTopPadding'] : $attr['contentVrPadding'];
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply includes/blocks/inline-notice/frontend.css.php]: Made changes as per suggestion.

File: src/blocks/inline-notice/styling.js [hunk 3]
```diff
@@ -149,6 +151,22 @@ function styling( attributes, clientId, name ) {
 	const posLeftUnitMob = ! isNaN( titleLeftPaddingMobile ) ? mobileTitlePaddingUnit : posLeftUnitTab;
 	const posRightUnitMob = ! isNaN( titleRightPaddingMobile ) ? mobileTitlePaddingUnit : posRightUnitTab;
 
+	// setting content padding Defaults
+	const contentPaddingTop = contentTopPadding? contentTopPadding : contentVrPadding;
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply src/blocks/inline-notice/styling.js]: Made changes as per discussion.

File: src/blocks/inline-notice/styling.js [hunk 4]
```diff
@@ -149,6 +151,22 @@ function styling( attributes, clientId, name ) {
...
+	const contentPaddingLeft = contentLeftPadding? contentLeftPadding : contentHrPadding;
+	const contentPaddingRight = contentRightPadding? contentRightPadding : contentHrPadding;
+	const contentPaddingBottom = contentBottomPadding? contentBottomPadding : contentVrPadding;
+
+	const contentPaddingTopMobile = contentTopPaddingMobile? contentTopPaddingMobile : contentVrPadding;
+	const conttentPaddingLeftMobile = contentLeftPaddingMobile? contentLeftPaddingMobile : contentHrPadding;
+	const contentPaddingRightMobile = contentRightPaddingMobile? contentRightPaddingMobile : contentHrPadding;
+	const contentpaddingBottomgMobile = contentBottomPaddingMobile? contentBottomPaddingMobile : contentVrPadding;
```
[the remark this answers is not in the document]
Comment (rajmohanreddy774) [reply src/blocks/inline-notice/styling.js]: Rectified.

File: README.md [hunk 5]
```diff
@@ -221,7 +221,8 @@ Our external packages use [Rating Star Component](https://github.com/n49/react-s
 * Improvement: Forms - Safely removed older style implementation for child blocks.
 * Improvement: Code - Improved Performance by optimizing apiFetch across the plugin.
 * Improvement: Code - Improved code quality for passing React props.
-
+* Fix: Inline Notice - Resolved issues wherein the content would have inconsistent default padding in the editor and the front-end, and removed the unwanted top margin on the front-end.
```
Comment (bsf-zanev) [inline README.md:224]: One minor change?

```suggestion
* Fix: Inline Notice - Resolved an issue wherein the content would have inconsistent default padding in the editor and the front-end.
* Fix: Inline Notice - Removed the unwanted top-margin in the front-end.
```

File: src/blocks/inline-notice/styling.js [hunk 6]
```diff
@@ -109,45 +129,61 @@ function styling( attributes, clientId, name, deviceType ) {
...
+		rPaddingMobile = titlePaddingRightMobile;
+		lPaddingTablet = titlePaddingLeftTablet;
+		rPaddingTablet = titlePaddingRightTablet;
 	}
 
 	// Setting the Icon Position Defaults
-	const posTopTab = ! isNaN( titleTopPaddingTablet ) ? titleTopPaddingTablet : titleTopPadding;
-	const posLeftTab = ! isNaN( titleLeftPaddingTablet ) ? titleLeftPaddingTablet : titleLeftPadding;
-	const posRightTab = ! isNaN( titleRightPaddingTablet ) ? titleRightPaddingTablet : titleRightPadding;
+	const posTopTab = ! isNaN( titlePaddingTopTablet ) ? titlePaddingTopTablet : titlePaddingTop;
```
Comment (bsf-zanev) [inline src/blocks/inline-notice/styling.js:156]: Should we follow the more precise number check as discussed over call for this as well?

```suggestion
	const posTopTab = ( 'number' === typeof titlePaddingTopTablet ) ? titlePaddingTopTablet : titlePaddingTop;
```

Same for all other places below.

**Review by bsf-zanev — COMMENTED** [review]
Just one minor suggestion and changelog update.

File: README.md (hunk 5 above)
Comment (rajmohanreddy774) [reply README.md]: Updated.

File: src/blocks/inline-notice/styling.js (hunk 6 above)
Comment (rajmohanreddy774) [reply src/blocks/inline-notice/styling.js]: Changes done.

**Review by bsf-zanev — APPROVED** [review]
Looks good to me, I'll see if we can combine the chagelog entries into a short single entry if possible.

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Review by adi3890 — APPROVED** [review]

**Pull request merged**